node-wasm/src/client.rs

```diff
@@ -405,19 +405,19 @@ impl WasmNodeConfig {
                 .pruning_delay(MIN_PRUNING_DELAY)
         };
 
-        builder = builder.network(network).sync_batch_size(128);
-
-        let mut bootnodes = Vec::with_capacity(self.bootnodes.len());
-
-        for addr in self.bootnodes {
-            let addr = addr
-                .parse()
-                .with_context(|| format!("invalid multiaddr: '{addr}"))?;
-            let resolved_addrs = resolve_dnsaddr_multiaddress(addr).await?;
-            bootnodes.extend(resolved_addrs.into_iter());
-        }
+        let bootnodes = self
+            .bootnodes
+            .into_iter()
+            .map(|addr| {
+                addr.parse()
+                    .with_context(|| format!("invalid multiaddr: {addr}"))
+            })
+            .collect::<Result<Vec<Multiaddr>, _>>()?;
 
-        builder = builder.bootnodes(bootnodes);
+        builder = builder
+            .network(network)
+            .sync_batch_size(128)
+            .bootnodes(bootnodes.into_iter());
```

Comment:
`into_iter()` is not needed.

```suggestion
            .bootnodes(bootnodes);
```